src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6034,6 +6007,144 @@ private bool IsOpenName()
+                var tmp = this.ParseQualifiedNameRight(NameOptions.InTypeList, explicitInterfaceName, separator);
+                Debug.Assert(!ReferenceEquals(tmp, explicitInterfaceName), "We should have consumed something and updated explicitInterfaceName");
+                explicitInterfaceName = tmp;
+
+                // Now, get the next separator.
+                if (this.CurrentToken.Kind == SyntaxKind.ColonColonToken)
+                {
+                    separator = this.EatToken();
+
+                    if (reportAnErrorOnMispacedColonColon)
+                    {
+                        // The https://github.com/dotnet/roslyn/issues/53021 is tracking fixing this in general
+                        separator = this.AddError(separator, ErrorCode.ERR_UnexpectedAliasedName);
+                    }
+
+                    separator = this.ConvertToMissingWithTrailingTrivia(separator, SyntaxKind.DotToken);
+                }
+                else
+                {
+                    separator = this.EatToken(SyntaxKind.DotToken);
+                }
+            }
+
+            _termState = saveTerm;
+        }
+
+        /// <summary>
+        /// This is an adjusted version of <see cref="ParseMemberName"/>.
+        /// When it returns true, it stops at operator keyword (<see cref="IsOperatorKeyword"/>).
+        /// When it returns false, it does not advance in the token stream.
+        /// </summary>
+        private bool IsOperatorStart(out ExplicitInterfaceSpecifierSyntax explicitInterfaceOpt, bool advanceParser = true)
+        {
+            explicitInterfaceOpt = null;
+
+            if (IsOperatorKeyword())
+            {
+                return true;
+            }
+
+            if (this.CurrentToken.Kind != SyntaxKind.IdentifierToken)
+            {
+                return false;
+            }
+
+            NameSyntax explicitInterfaceName = null;
+            SyntaxToken separator = null;
+
+            ResetPoint beforeIdentifierPoint = GetResetPoint();
+
+            try
+            {
+                while (true)
+                {
+                    // now, scan past the next name.  if it's followed by a dot then
+                    // it's part of the explicit name we're building up.  Otherwise,
+                    // it should be an operator token
+                    var point = GetResetPoint();
+                    bool isPartOfInterfaceName;
+                    try
+                    {
+                        ScanNamedTypePart();
+                        isPartOfInterfaceName = IsDotOrColonColon();
+                    }
+                    finally
+                    {
+                        this.Reset(ref point);
+                        this.Release(ref point);
+                    }
+
+                    if (!isPartOfInterfaceName)
+                    {
+                        // We're past any explicit interface portion
```

Comment:
It feels like we could make the parse for `OperatorDeclaration_ExplicitImplementation_04` significantly better if we checked to see if the current token was `operator` at this point (back in the try/finally, more likely). The parse and resulting errors aren't particularly great for a piece of code that just missing a `.`.

Reply:
Hmm, actually I suppose it's not that simple given that we don't know if it was supposed to be a return type at this point. I'm still wondering if we can improve that error case, but it's not this simple.

Reply:
> It feels like we could make the parse for OperatorDeclaration_ExplicitImplementation_04 significantly better if we checked to see if the current token was operator at this point (or back in the try/finally, more likely). The parse and resulting errors aren't particularly great for a piece of code that's just missing a ..

Could you please provide more details about what you think is happening, what you are proposing to do and how do you think it is going to affect the test scenario? BTW, I believe the current token is not going to be an `operator` , I think it will be the token following the last `.`.

Reply:
> Hmm, actually I suppose it's not that simple given that we don't know if it was supposed to be a return type at this point. I'm still wondering if we can improve that error case, but it's not this simple.

I am willing to experiment with this scenario separately.

Reply:
It looks like, for cases like `int N.I operator` (coming from line 2339), we know that we have a return type already, so if we encounter an `operator` token this must have been intended as an explicit implementation. As part of the try/finally, we could additionally check `IsOperatorKeyword()`, and if it is and we know this must be an explicit qualification we just report a missing `.` token and construct a tree that is otherwise identical to the success case. I imagine this should get rid of almost all the errors in that test case and leave us with a parse tree that doesn't assume `N.I` was a method name and that the parens were part of a tuple type.